src/tools/schema2base.ts

```diff
@@ -7,75 +7,108 @@
  * subject to an additional IP rights grant found at
  * http://polymer.github.io/PATENTS.txt
  */
-
 import fs from 'fs';
-import minimist from 'minimist';
 import path from 'path';
+import minimist from 'minimist';
+import {Schema} from '../runtime/schema.js';
 import {Manifest} from '../runtime/manifest.js';
-
-
-export function typeSummary(descriptor) {
-  switch (descriptor.kind) {
-    case 'schema-primitive':
-      return `schema-primitive:${descriptor.type}`;
-
-    case 'schema-collection':
-      return `schema-collection:${descriptor.schema.type}`;
-
-    default:
-      return descriptor.kind;
+import {Dictionary} from '../runtime/hot.js';
+
+export abstract class Schema2Base {
+  constructor(readonly opts: minimist.ParsedArgs) {}
+
+  async call() {
+    fs.mkdirSync(this.opts.outdir, {recursive: true});
+    for (const file of this.opts._) {
+      if (fs.existsSync(file)) {
+        await this.processFile(file);
+      } else {
+        throw new Error(`File not found: ${file}`);
+      }
+    }
   }
-}
 
+  private async processFile(src: string) {
+    const outName = this.outputName(path.basename(src));
+    const outPath = path.join(this.opts.outdir, outName);
+    console.log(outPath);
+    if (this.opts.update && fs.existsSync(outPath) && fs.statSync(outPath).mtimeMs > fs.statSync(src).mtimeMs) {
+      return;
+    }
 
-export class Schema2Base {
-  private desc: string;
-  private fileNamer: (schemaName: string) => string;
-  private generate: (name: string, schema) => string;
-
-  constructor(description: string, fileNamer: (schemaName: string) => string, generator: (name: string, schema) => string) {
-    this.desc = description;
-    this.fileNamer = fileNamer;
-    this.generate = generator;
-  }
+    const contents = fs.readFileSync(src, 'utf-8');
+    const manifest = await Manifest.parse(contents);
 
+    // Collect declared schemas along with any inlined in particle connections.
+    const schemas: Dictionary<Schema> = {...manifest.schemas};
```

Comment:
I would break out this section into a function in the class.

Reply:
Ack. I generally prefer not to make single-use mini-functions when the surrounding one isn't very big itself.